packages-content-model/roosterjs-content-model-editor/lib/modelApi/common/mergeModel.ts

```diff
@@ -120,14 +120,21 @@ export function mergeModel(
 function mergeParagraph(
     markerPosition: InsertPoint,
     newPara: ContentModelParagraph,
-    mergeToCurrentParagraph: boolean
+    mergeToCurrentParagraph: boolean,
+    option: MergeModelOption | undefined
 ) {
     const { paragraph, marker } = markerPosition;
     const newParagraph = mergeToCurrentParagraph
         ? paragraph
         : splitParagraph(markerPosition, newPara.format);
     const segmentIndex = newParagraph.segments.indexOf(marker);
 
+    if (option?.mergeFormat == 'none' && mergeToCurrentParagraph) {
+        newParagraph.segments.forEach(segment => {
+            segment.format = { ...(newParagraph.segmentFormat || {}), ...segment.format };
```

Comment:
According how format is working on segment today, existing segments should already have a copy of segment format from the paragraph. That means even we don't rewrite format on segments, it should still keep existing formats. (of cause for the new merged segments, we still need to copy the format).

Reply:
I see, so, we only need to remove the segment format from the paragraph then, right? 
Since we already set the segment (default) format from container paragraph to each child segment's format.

Reply:
For those newly merged segments, we still need to merge the format since there may be some format specified in block.segmentFormat but not exist in new segment. For those existing segments, nothing need to be done.

But I think to simplify the code, it should be fine to just do merge for all segments like you already did.